graphql_compiler/cost_estimation/analysis.py

```diff
@@ -53,9 +54,81 @@ def _convert_int_interval_to_field_value_interval(
     return Interval(lower_bound, upper_bound)
 
 
+def get_single_field_filters(
+    schema_info: QueryPlanningSchemaInfo, query_metadata: QueryMetadataTable,
+) -> Dict[PropertyPath, Set[FilterInfo]]:
+    """Find the single field filters for each field.
+
+    Filters that apply to multiple fields, like name_or_alias, are ignored.
+    Filters inside fold scopes are not considered.
+
+    Args:
+        schema_info: QueryPlanningSchemaInfo
+        query_metadata: info on locations, inputs, outputs, and tags in the query
+
+    Returns:
+        A set of filters for each field.
+    """
+    single_field_filters = {}
+    for location, location_info in query_metadata.registered_locations:
+        if not isinstance(location, Location):
+            continue  # We don't paginate inside folds.
```

Comment:
Can you remove this block and remove the "filters inside fold scopes are not considered" bit from the documentation? We already perform this check in `get_fields_eligible_for_pagination`. I don't see why we need it here and it makes the function docstring more complicated.

Reply:
The function is not called `get_single_field_filters_for_pagination`, and in fact it could be used by cost estimation if it did look inside fold scopes.

Reply:
I would extend it to include folded locations, but it's not as simple as removing this block.